packages/amplify-prompts/src/progressbars/multibar.ts

```diff
@@ -0,0 +1,240 @@
+          // We do not update the item status if it has already failed.
+          if (item.status !== updateObj.payload.ResourceStatus && !itemFailure) {
+            barToUpdate.updateItem(updateObj.name, updateObj.payload);
+            this.updated = true;
+          }
+        } else {
+          barToUpdate.addItem(updateObj.name, updateObj.payload);
+          this.updated = true;
+        }
+        // Since items can have multiple finished states (CloudFormation nuance),
+        // we do not increment the progress bar if it has already failed/succeeded.
+        if (this.options.itemCompleteStatus.includes(updateObj.payload.ResourceStatus) && !finishedStatus && !itemFailure) {
+          barToUpdate.increment();
+          this.updated = true;
+        }
+      }
+    }
+
+    /**
+     * Increments value of a bar indexed by name
+     */
+    incrementBar(name: string, value: number) : void {
+      const barDetails = this.getBar(name);
+      if (barDetails) {
+        const barToUpdate = barDetails.bar;
+        barToUpdate.increment(value);
+        this.updated = true;
+      }
+    }
+
+    /**
+     * Finishes a bar indexed by name
+     */
+    finishBar(name: string) : void {
+      const barDetails = this.getBar(name);
+      if (barDetails) {
+        const barToUpdate = barDetails.bar;
+        if (!barToUpdate.hasFinished() && !barToUpdate.hasFailed()) {
+          barToUpdate.finish();
+          this.updated = true;
+        }
+      }
+    }
+
+    /**
+     * Creates a set of progress bars under the multi bar
+     */
+    create(bars: {
+        name: string,
+        value: number,
+        total: number,
+        payload: ProgressPayload
+    }[]) : void {
+      if (!this.bars.length) {
+        this.terminal.newLine();
+        if (this.options.hideCursor === true) {
+          this.terminal.cursor(false);
+        }
+        this.isActive = true;
+      }
+      bars.forEach(config => {
+        const newBar = new Bar(this.options);
+        newBar.start(config.total, config.value, config.payload);
+        this.bars.push({ name: config.name, bar: newBar });
+        this.count += 1;
+      });
+      if (bars.length) {
+        this.updated = true;
+        this.render();
+      }
+      this.count += bars.length;
+    }
+
+    /**
+     * Returns count of progress bars under the multi bar
+     */
+    getBarCount() : number {
+      return this.count;
+    }
+
+    /**
+     * Stop all progress bars under the multi bar
+     */
+    stop() : void {
+      this.isActive = false;
+      clearTimeout(this.timer);
+
+      // Change prefix text according to success/failure
+      let initLine : string = chalk.green(this.options.successText || '');
+      for (let i = 0; i < this.bars.length; i++) {
+        if (this.bars[i].bar.hasFailed()) {
+          initLine = chalk.red(this.options.failureText || '');
+          break;
+        }
+      }
```

Comment:
nit: I prefer a for...of loop

```suggestion
      for (const { bar } of this.bars) {
        if (bar.hasFailed()) {
          initLine = chalk.red(this.options.failureText || '');
          break;
        }
      }
```